app/controllers/hosts_controller.rb

```diff
@@ -546,7 +546,8 @@ def process_taxonomy
   end
 
   def template_used
-    host      = Host.new(clean_interfaces_attributes.except(:host_parameters_attributes))
+    host = params[:id] ? Host.find(params[:id]) : Host.new
```

Comment:
> To be more safe for associations added in future we could ignore all /.*_ids$/ keys, not sure if it's any better.

This might be safer in case plugins are involved - if the host form has been extended any to include nested attributes, then plugin authors aren't going to have much control over the .except() call in the line below.  It might be safer for everybody to filter them out by default.

Would using .readonly help?